### DISCUSSION TITLE
Error when running compiled (Using Mingw-46 under MSYS2) Magick++ C++ executable windows

### DISCUSSION THREAD

**Discussion by b-rad15** [ImageMagick/ImageMagick#3099] [Help] [answered]
To start off, the compilation and execution work fine on Linux, so the code is probably not the issue, was hoping to get windows executables built for distribution purposes

I installed the ImageMagick libraries via [this package](https://packages.msys2.org/package/mingw-w64-x86_64-imagemagick?repo=mingw64) (outdated but should still work fine and there isn't a newer packaged version)
then I built it using the command (my other build stuff removed) g++ ... \`Magick++-config --cppflags --cxxflags --ldflags --libs\` and that command spits out 
```
-IC:/msys64/mingw64/include/ImageMagick-7 -fopenmp -DMAGICKCORE_HDRI_ENABLE=1 -DMAGICKCORE_QUANTUM_DEPTH=16 -D_DLL -D_MT
-IC:/msys64/mingw64/include/ImageMagick-7 -fopenmp -DMAGICKCORE_HDRI_ENABLE=1 -DMAGICKCORE_QUANTUM_DEPTH=16 -D_DLL -D_MT
-LC:/msys64/mingw64/lib -lMagick++-7.Q16HDRI -lMagickWand-7.Q16HDRI -lMagickCore-7.Q16HDRI
-LC:/msys64/mingw64/lib -lMagick++-7.Q16HDRI -lMagickWand-7.Q16HDRI -lMagickCore-7.Q16HDRI
```
as the result (I have also tried without cxxflags and libs so only one of each is printed but that doesn't help either)

the build works and creates an executable and there are no errors or warnings but once I try to run that executable in the MSYS2 command line this error shows up `error while loading shared libraries: libMagickCore-7.Q16HDRI-7.dll: cannot open shared object file: No such file or directory` there is a file named libMagickCore-7.Q16HDRI-7.dll.a in mingw's lib dir but even copying that to the working directory doesn't fix the issue. Renaming that to libMagickCore-7.Q16HDRI-7.dll and placing it in the working dir causes either an illegal instruction error to show up or nothing and the app crashes. gdb was no help finding the error but it does happen before a breakpoint in main could be triggered.

Any idea what the issue may be? I'm building from sources now to see if that fixes it but I don't expect that it will. I'll update once I test with that but it's gonna be a little bit.

**Comment by urban-warrior** [comment, marked as answer]
What does `ldd /mingw64/bin/convert` reveal?  It will tell you where it is looking to find libMagickCore-7.Q16HDRI-7.dll.  Search for libMagickCore-7.Q16HDRI-7.dll on your system.  Where is located?  It likely would work if its in the `/mingw64/bin` folder.

**Reply by b-rad15** [reply]
Just tried that now, the library was in `/mingw64/bin` so not sure why it didn't load. After copying all the needed dlls to the same folder as the executable the file ran "fine" (error from before where it gave no output even though the first line of the main function writes to stdout seems to be due to a failed image load that happened before the main function). Thanks!
